Feature/floating maker changes

ricardojmendez opened the pull request:
Multiple changes to floatingMaker, including adding configuration options, avoiding crashes on exceptions, and handling issues #65  and #66.

0xbigz commented on src/config.ts:
```diff
@@ -44,7 +52,7 @@ export type BotConfigMap = {
-	trigger?: BaseBotConfig;
+	trigger?: FloatingMakerConfig;
 	liquidator?: LiquidatorConfig;
 	floatingMaker?: BaseBotConfig;
```
shouldnt this be for `floatingMaker` not `trigger`?

ricardojmendez replied:
Ugh, you are correct.

0xbigz commented on src/bots/floatingMaker.ts:
```diff
@@ -296,53 +331,124 @@ export class FloatingPerpMakerBot implements Bot {
+			} catch (e) {
+				console.log('Error placing new orders');
+				console.error(e);
+				console.log('----');
```
what error are you typically hitting that leads to this?

ricardojmendez replied:
Usually it seems to be "Order already processed", but I've seen the rate timeout.

0xbigz commented on src/bots/floatingMaker.ts:
```diff
@@ -296,53 +331,124 @@ export class FloatingPerpMakerBot implements Bot {
+								.getUserAccount()!
+								.authority.toBase58()}-${o.orderId}`
+						);
+						return this.driftClient.cancelOrder(o.orderId);
```
theres now a `cancelOrders` instruction that will be most compute/tx efficient

ricardojmendez replied:
Thanks, I hadn't checked the options. I'm going to use `cancelOrdersByIds`, since I've been thinking it's best to only cancel the orders with a price offset (those the bot placed), in case someone wants to de-risk using a stop.

ricardojmendez replied:
I will use cancelOrders on bot start to clean up any leftovers.

I think overall it would be more fitting to refactor `updateOpenOrdersForMarket` into two functions: one cancels all orders across that needs canceling in a single call, across all markets, and another takes the new order calculation/placement that `updateOpenOrdersForMarket` currently does.

Leaving that aside for now to avoid bloating this PR and making a note of it.

0xbigz commented on src/bots/floatingMaker.ts:
```diff
@@ -296,53 +331,124 @@ export class FloatingPerpMakerBot implements Bot {
+		const orderSizeShort =
+			exposureDir == PositionDirection.SHORT
+				? orderSizeCurrent
+				: orderSizeOther;
```
do you think it'd be better/safer if `orderSize` was specified in quote and then ceil'd to the base step size?

ricardojmendez replied:
Do you mean as an API option?

ricardojmendez commented:
Fixed some issues, and added a few commits, including implementing bias order skewing and handling positions in a "closing" state, as well as cleaning up outdated comments.

ricardojmendez commented:
Closing, since the PR is old and the code has drifted. I may re-submit after a review.
